server/src/main/java/io/spine/server/aggregate/AggregateCommandEndpoint.java

```diff
@@ -46,13 +55,51 @@ I handle(AggregateRepository<I, A> repository, CommandEnvelope command) {
         return endpoint.handle();
     }
 
-
     static <I, A extends Aggregate<I, ?, ?>>
     AggregateCommandEndpoint<I, A>
     of(AggregateRepository<I, A> repository, CommandEnvelope command) {
         return new AggregateCommandEndpoint<>(repository, command);
     }
 
+    @Override
+    protected void deliverNowTo(I aggregateId) {
+        ensureNotDuplicate(aggregateId);
```

Comment:
This means that we read events twice: once when checking, and another time when actually loading the aggregate. It's an expensive operation. We need to check it differently. Maybe query recent history of an aggregate after it's already loaded?